src/ImageSharp/Processing/Processors/Normalization/AdaptiveHistogramEqualizationProcessor{TPixel}.cs

```diff
@@ -244,12 +238,12 @@ protected override void OnFrameApply(ImageFrame<TPixel> source)
             {
                 for (int dy = yStart; dy < yEnd; dy++)
                 {
-                    int dyOffSet = dy * sourceWidth;
+                    Span<TPixel> rowSpan = source.GetPixelRowSpan(dy);
                     int tileX = 0;
                     int xLimit = Math.Min(x + tileWidth, sourceWidth - 1);
                     for (int dx = x; dx < xLimit; dx++)
```

Comment:
I think if we sliced the `rowSpan` and used `.Length` then we should be able avoid all bounds checks and maintain previous performance in a safe manner on all targets.